kill_hogs/kill_hogs.py

```diff
@@ -123,7 +123,21 @@ def is_restricted(username: str, pattern: str = '^(?!root).*'):
     return re.match(pattern, username) is not None
 
 
+def proc_is_using_gpu(pid):
```

Comment:
please use the new type hints. They are used elsewhere in this script and provide more useful error messages.
https://realpython.com/lessons/type-hinting/